src/components/Button/Button.tsx

```diff
@@ -82,6 +82,7 @@ function ButtonComponent({
       theme={theme}
       type={type}
       variant={variant}
+      data-id={props.dataId}
```

Comment:
This shouldn't be necessary because {...props} are spread to the button root. Adding data-id does make it more explicit, but I'd like to support the pattern of being able to add any data-* attribute you want to.